### ISSUE TITLE
`btrbk snapshot list` fails to find snapshots if they are in a separately-mounted btrfs volume.

### ISSUE THREAD

**Issue by g-rocket** [digint/btrbk#333]
I have a separate volume for snapshots, which is mounted under the main volume I am backing up.  Most things work fine, but `btrbk snapshot list` finds 0 snapshots, because it appears to be looking only for snapshots that are direct subvolumes of the volume being backed up -- it is ignoring my `snapshot_dir` directive.

Config (`/etc/btrbk/btrbk.conf`):
```
transaction_syslog         daemon
snapshot_create            always
incremental                yes
timestamp_format           long-iso
preserve_hour_of_day       5
preserve_day_of_week       sunday
snapshot_preserve_min      1d
target_preserve_min        no
target_preserve            48h 20d 10w 24m *y
stream_compress            no
lockfile                   /var/lock/btrbk.lock
snapshot_dir               .snapshots
target /mnt/storage/backups/microlith
volume /
  subvolume .
volume /home
  subvolume .
  subvolume gaviny/iot

```

Filesystem layout (`/etc/fstab`):
```
LABEL=system        	/         	btrfs     	rw,noatime,compress=lzo,ssd,space_cache,subvolid=256,subvol=/root,subvol=root	0 0
LABEL=EFI           	/boot     	vfat      	rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,utf8,errors=remount-ro	0 2
LABEL=system        	/home     	btrfs     	rw,noatime,compress=lzo,ssd,space_cache,subvolid=257,subvol=/home,subvol=home	0 0
LABEL=system        	/.snapshots	btrfs     	rw,noatime,compress=lzo,ssd,space_cache,subvolid=258,subvol=/snapshots,subvol=snapshots	0 0
LABEL=system        	/home/.snapshots	btrfs     	rw,noatime,compress=lzo,ssd,space_cache,subvolid=258,subvol=/snapshots,subvol=snapshots	0 0
/dev/mapper/swap          	none      	swap      	defaults  	0 0
LABEL=storage           /mnt/storage    btrfs           rw,noatime,compress=lzo 0 0
```

**Comment by digint** [comment]
According to the man-page, `btrbk snapshot list` should "List all snapshots (and corresponding backups). Note that all snapshots related to configured subvolumes are listed, not only the ones created by btrbk.".

Actually it lists all snapshots below the root, but not taking mounts into account.

**Comment by digint** [comment]
Fixed in: 31475a3

Included in [v0.31.0](/digint/btrbk/releases/tag/v0.31.0)
